hmp/models/base.py

```diff
@@ -0,0 +1,205 @@
+"""Models to estimate event probabilities."""
+import gc
+import itertools
+import multiprocessing as mp
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from itertools import cycle, product
+from typing import Any
+from warnings import resetwarnings, warn
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from pandas import MultiIndex
+from scipy.signal import correlate
+from scipy.stats import norm as norm_pval
+
+from hmp.trialdata import TrialData
+
+try:
+    __IPYTHON__
+    from tqdm.notebook import tqdm
+except NameError:
+    from tqdm import tqdm
+
+
+@dataclass
+class EventProperties():
+    sfreq: float
+    steps: float
+    shape: float
```

Comment:
I understand why you made it so but in the end shape is an argument for the distribution. Do you think we can build the distribution here in the class instead of BaseModel?

Reply:
I was planning to create a distribution class, but @maartenschermer might be busy with that?